src/modules/users/components/forms/security-form/security-form.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Fragment } from 'react/jsx-runtime';
+import { useNavigate } from 'react-router-dom';
+import Box from '@mui/material/Box';
+import Breadcrumbs from '@mui/material/Breadcrumbs';
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+import Button from '@mui/material/Button';
+import { type ProfileFormData, type FormDataKeys, type AxiosErrorResponse } from '#/modules/users/types';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { AxiosError } from 'axios';
+import { ROUTES } from '#/modules/users/constants';
+import { ENDPOINTS as AUTH_ENDPOINTS } from '#/modules/auth/constants';
+import { api } from '#/configs';
+import { DeleteAccountModal, ErrorMessage } from '#/modules/users/components';
+import { getFirstErrorMessage } from '#/utils';
+import { ConfirmPasswordInput, OldPasswordInput, PasswordInput } from '../../input-fields';
+import { HeaderComponent } from '#/modules/home/components/header';
+import { toast } from 'react-toastify';
+import { useQueryClient } from '@tanstack/react-query';
+import { UpdatePasswordSchema } from '#/zod';
+import { securityFormStyles as styles } from './security-form.styles';
+
+export function SecurityForm() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setError,
+  } = useForm<ProfileFormData>({ resolver: zodResolver(UpdatePasswordSchema) });
+  const [focusedField, setFocusedField] = useState<FormDataKeys | null>(null);
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+
+  function handleClick() {
+    navigate(ROUTES.accountSettings);
+  }
+
+  function handleFocus(field: FormDataKeys) {
+    setFocusedField(field);
+  }
+
+  function handleBlur() {
+    setFocusedField(null);
+  }
+
+  function onSubmit(data: ProfileFormData) {
+    api
+      .post(`${AUTH_ENDPOINTS.updatePassword}`, { oldPassword: data.oldPassword, newPassword: data.password })
+      .then(() => {
+        queryClient.invalidateQueries({ queryKey: ['auth-user'] });
+        toast('Success');
+
+        navigate(ROUTES.root);
+      })
+      .catch((err) => {
+        if (err instanceof AxiosError) {
+          const errorData = err.response?.data as AxiosErrorResponse;
+          setError('firstName', { message: errorData.message });
+        } else {
+          setError('firstName', { message: 'Something went wrong' });
```

Comment:
Same comment about error field name. Especially since this is a form to update the password, there is no first name.